server/py/services/api/initial_data.py

```diff
@@ -993,6 +998,47 @@ def _create_project_summaries(db, db_session):
     db._upsert(db_session, project_summaries, ignore=True)
 
 
+def _init_system_id(db_session: sqlalchemy.orm.Session):
+    """
+    Initializes a system id for MLRun deployment.
+    The system id is first checked in the database. If it does not exist, the function checks for a configured id
+    (from environment variables or config), and if neither is found, a new random one is generated and stored.
+    """
+
+    db = framework.db.sqldb.db.SQLDB()
+
+    # check if a system id already exists in the database
+    system_id = db.get_system_id(db_session)
+
+    if system_id is not None:
+        logger.info("Existing system id found in the database", system_id=system_id)
+    else:
+        # check if the system id is already set in the config
+        system_id = _get_configured_system_id()
+
+        if system_id is not None:
+            logger.info("Using configured system id", system_id=system_id)
+            db.store_system_id(db_session, system_id)
+        else:
+            # if no system id is found, generate a new one
+            system_id = _generate_system_id()
+            db.store_system_id(db_session, system_id)
+
+    # set the system id in mlrun config
+    mlrun.mlconf.system_id = system_id
+
+
+def _get_configured_system_id() -> typing.Optional[str]:
+    return mlrun.mlconf.system_id or None
```

Comment:
More a usability + validation question. 

if we say that system_id is base64 string of len==6, 

1. Should we enforce this and validate what user passed? 
2. Should we translate users input to base64(because current logic just takes it as is), or how else can we check that it's base64?

Reply:
1. the base64 string and len of 6 are used only when generating a new random one.. ( if it does not exist in the db )
2. users will not use this id, as it comes with MLRun deployment, and should be changed only by support ( by deleting it from the db which will cause generating a new one on next installation, or by config )

Reply:
@Yacouby well, what happens if support mistakenly sets in to 7, 8 or 10 symbols?

Reply:
@Yacouby @rokatyy i think the 32bit number was chosen more for its high probability for uniqueness, rather them it's actual length.
The custom ID is a nice-to-have and I don't think we should really limit it# perhaps maybe by coomin name length limitations (such as 63 chats for k8s resources).
We can bring @theSaarco into the discussion to validate the original meaning.

Reply:
Even if we do decide to limit the size, it can be done in a followup PR